source/base/Shmoose.cpp

```diff
@@ -250,7 +250,8 @@ void Shmoose::sendMessage(QString const &message, QString const &type)
 
 void Shmoose::sendFile(QString const &toJid, QString const &file)
 {
-    if (httpFileUploadManager_->requestToUploadFileForJid(file, toJid) == false)
+    if (httpFileUploadManager_->requestToUploadFileForJid(file, toJid, 
+                                                          lurchAdapter_->isOmemoUser(toJid) && (! settings_->getSendPlainText().contains(toJid))))
```

Comment:
Please store this
'lurchAdapter_->isOmemoUser(toJid) && (! settings_->getSendPlainText().contains(toJid))' to a local 'bool shouldEncryptFile' and use that bool in the call. 
It is very hard to read and understand currently.

Reply:
And is the logic changed? The original call checks for ' == false' to output the qDebug msg. Now it must equal to true?

Reply:
Let me check.

Reply:
Fixed